factory/state/stateComponents.go

```diff
@@ -55,37 +52,18 @@ type stateComponents struct {
 
 // NewStateComponentsFactory will return a new instance of stateComponentsFactory
 func NewStateComponentsFactory(args StateComponentsFactoryArgs) (*stateComponentsFactory, error) {
-	if args.Core == nil {
+	if check.IfNil(args.Core) {
 		return nil, errors.ErrNilCoreComponents
 	}
-	if check.IfNil(args.Core.Hasher()) {
-		return nil, errors.ErrNilHasher
-	}
-	if check.IfNil(args.Core.InternalMarshalizer()) {
-		return nil, errors.ErrNilMarshalizer
-	}
-	if check.IfNil(args.Core.PathHandler()) {
-		return nil, errors.ErrNilPathHandler
-	}
-	if check.IfNil(args.ShardCoordinator) {
-		return nil, errors.ErrNilShardCoordinator
-	}
 	if check.IfNil(args.StorageService) {
 		return nil, errors.ErrNilStorageService
 	}
-	if check.IfNil(args.ChainHandler) {
```

Comment:
Not needed to check for nil ChainHandler?

Reply:
only kept checks on the components that are used inside the factory and removed the ones that are only forwarded to subcomponents(which should do their checks on the received params)

example:
- chainHandler is saved internally only to be forwarded
   - [1] https://github.com/multiversx/mx-chain-go/blob/more_factory_tests_p2/factory/state/stateComponents.go#L144
   - [2] https://github.com/multiversx/mx-chain-go/blob/more_factory_tests_p2/factory/state/stateComponents.go#L149
- core components is saved to forward subcomponents, which requires the check